pkgs/top-level/all-packages.nix

```
@@ -16628,7 +16628,7 @@ with pkgs;
       /**/ if platform.isDarwin then 16
       else if platform.isFreeBSD then 12
       else if platform.isAndroid then 12
-      else if platform.isLinux then 16
+      else if platform.isLinux then 17
       else if platform.isWasm then 16
```

wasi-sdk/wasilibc 21 should support LLVM 17 as they use it by default.

```suggestion
      else if platform.isWasm then 17
```

---

I'd like to leave that as is, since it works fine on 16, and there's no reason introducing another version on Darwin.  In general we're a bit confused about whether this is supposed to be minimum versions like the comment says, or default versions.